src/hotspot/share/code/dependencies.hpp

```diff
@@ -103,6 +103,9 @@ class Dependencies: public ResourceObj {
   // type now includes N, that is, all super types of N.
   //
   enum DepType {
+    // _type is initially set to -1, to prevent "already at end" assert
+    undefined_dependency = -1,
```

Comment:
Preserving the existing value seems file, though it appears any value >= TYPE_LIMIT would work just as well.

Reply:
FWIW, I was also entertaining the idea of a solution like that with the hope that it would fit better with the usage in the iterators.